tools/refguide_check.py

```diff
@@ -790,37 +792,23 @@ def _run_doctests(tests, full_name, verbose, doctest_warnings):
     runner = DTRunner(full_name, checker=Checker(), optionflags=flags,
                       verbose=verbose)
 
-    output = []
+    output = io.StringIO()
     success = True
-    def out(msg):
-        output.append(msg)
-
-    class MyStderr:
-        """
-        Redirect stderr to the current stdout
-        """
-        def write(self, msg):
-            if doctest_warnings:
-                sys.stdout.write(msg)
-            else:
-                out(msg)
+    tmp_stderr = sys.stdout if doctest_warnings else output
```

Comment:
I believe this mirrors the old behavior of 
```
    if doctest_warnings:
        sys.stdout.write(msg)
    else:
        out(msg)
```

Reply:
Yeah, this looks good to me